src/daemon/common.c

```diff
@@ -1668,3 +1672,32 @@ void strarray_free (char **array, size_t array_len) /* {{{ */
 		sfree (array[i]);
 	sfree (array);
 } /* }}} void strarray_free */
+
+#ifdef HAVE_SYS_CAPABILITY_H
+int check_capability (int capability) /* {{{ */
+{
+#ifdef _LINUX_CAPABILITY_VERSION_3
+	struct __user_cap_header_struct cap_header_data;
```

Comment:
I'm feeling very uncomfortable using private system structures. I think we have two options:

1.  Allocate memory on the heap, using `calloc(sizeof (*cap_header), 1)`. Unfortunately, I'm not aware of a stack-only way of initializing pointer types :\
2.  Use `cap_get_proc()` and `cap_get_flag()` (and `cap_free()`) instead of the Linux-only interface. This seems to be the nicer interface to me.